### ISSUE TITLE
@Configuration && @Logging not injecting values on Assembler

### ISSUE THREAD

**Issue by Sherpard** [seedstack/business#80]
Trying to get a configuration value from configuration file does nothing when done on an Assembler. even when other objects are correctly injected (Such an FluentAssembler), but it is Loaded on the Resource

How to Reproduce

TestResource as entry point with @GET method, 

`XNIO-1 task-1 c.c.basecamp.rest.TestResource Configuration is: http://127.0.0.1`

Resource Injects a FinderInterface that is implemented on a JpaFinder
`@Finder
public interface RepositoryFinder`
`public class RepositoryJpaFinder implements RepositoryFinder`

At this point @Configuration and @Logging are correctly injected
`XNIO-1 task-1 c.c.b.a.i.j.f.RepositoryJpaFinder Configuration is: http://127.0.0.1`

This finder injects a Repository (Repository is actually an entity on my domain, not a weird misconception)
`public interface RepositoryRepository extends GenericRepository<Repository, Long>
public class RepositoryJpaRepository extends BaseJpaRepository<Repository, Long>implements RepositoryRepository`

On this point, @Configuration and @Logging still working as intended
`XNIO-1 task-1 c.c.b.a.i.j.r.RepositoryJpaRepository Configuration is: http://127.0.0.1`

Finder converts Aggregate to Dto using a Custom Assembler, that inherits ModelMapper.
`public class RepositoryAssembler extends ModelMapperAssembler<Repository, RepositoryRepresentation>`

That Assembler injects a FluentAssembler right with @Inject, but @Logging and @Configuration does not bind anything. leaving both fields null

Assembler -> Configuration is null
Assembler -> Logger is null!

**Comment by adrienlauer** [comment]
I reproduced the problem in an integration test. The `configure*()` methods are called from within the constructor of `ModelMapperAssembler` and as such field injections have not happened yet.

I'm currently fixing that by delaying the `configure*()` methods until the first assemble/merge call. 

Meanwhile, as a workaround, you can specify the `@Inject` annotation on a constructor of you assembler and inject anything there. Since you cannot use the `@Configuration` annotation on a parameter you can inject `Application` instead and use the programmatic access to configuration:

```
public class MyAssembler extends ModelMapperAssembler<...> {

    @Inject
    public MyAssembler(Application application) {
        String test = application.getConfiguration().getString("test", "defaultValue");
    }

    ...  
}
```

**Comment by adrienlauer** [comment]
@Sherpard: aside from this issue, for your new project, I recommend that you use the REST project template: http://seedstack.org/getting-started/project-templates/rest.

If you frontend is done with the SeedStack Web framework, you can generate one with the pure Web template: http://seedstack.org/getting-started/project-templates/w20/.

To bridge the two you can rely on the W20 bridge add-on: http://seedstack.org/addons/w20-bridge/

**Comment by adrienlauer** [comment]
The fix is available in version 2.3.4 of the business framework. It should be available shortly on Maven central. To use this version, just override the business-\* dependencies in your project root pom with a dependency mangement **below** the seedstack-bom import:

```
<dependencyManagement>
    <dependencies>
        <dependency>
            <groupId>org.seedstack</groupId>
            <artifactId>seedstack-bom</artifactId>
            <version>16.7</version>
            <type>pom</type>
            <scope>import</scope>
        </dependency>
        <dependency>
            <groupId>org.seedstack.business</groupId>
            <artifactId>business-specs</artifactId>
            <version>2.3.4</version>
        </dependency>
        <dependency>
            <groupId>org.seedstack.business</groupId>
            <artifactId>business-core</artifactId>
            <version>2.3.4</version>
        </dependency>
    </dependencies>
</dependencyManagement>
```
